diff -Naur busybox-1.18.5.orig/include/libbb.h busybox-1.18.5/include/libbb.h
--- busybox-1.18.5.orig/include/libbb.h	2011-06-12 09:41:51.000000000 -0600
+++ busybox-1.18.5/include/libbb.h	2011-07-27 14:23:18.377240536 -0600
@@ -44,6 +44,12 @@
 #ifdef HAVE_SYS_STATFS_H
 # include <sys/statfs.h>
 #endif
+/* Don't do this here:
+ * #include <sys/sysinfo.h>
+ * Some linux/ includes pull in conflicting definition
+ * of struct sysinfo (only in some toolchanins), which breaks build.
+ * Include sys/sysinfo.h only in those files which need it.
+ */
 #if ENABLE_SELINUX
 # include <selinux/selinux.h>
 # include <selinux/context.h>
@@ -100,25 +106,6 @@
 #if !defined __FreeBSD__
 char *dirname(char *path);
 #endif
-/* Include our own copy of struct sysinfo to avoid binary compatibility
- * problems with Linux 2.4, which changed things.  Grumble, grumble. */
-struct sysinfo {
-	long uptime;			/* Seconds since boot */
-	unsigned long loads[3];		/* 1, 5, and 15 minute load averages */
-	unsigned long totalram;		/* Total usable main memory size */
-	unsigned long freeram;		/* Available memory size */
-	unsigned long sharedram;	/* Amount of shared memory */
-	unsigned long bufferram;	/* Memory used by buffers */
-	unsigned long totalswap;	/* Total swap space size */
-	unsigned long freeswap;		/* swap space still available */
-	unsigned short procs;		/* Number of current processes */
-	unsigned short pad;			/* Padding needed for m68k */
-	unsigned long totalhigh;	/* Total high memory size */
-	unsigned long freehigh;		/* Available high memory size */
-	unsigned int mem_unit;		/* Memory unit size in bytes */
-	char _f[20 - 2 * sizeof(long) - sizeof(int)]; /* Padding: libc5 uses this.. */
-};
-int sysinfo(struct sysinfo* info);
 #ifndef PATH_MAX
 # define PATH_MAX 256
 #endif
diff -Naur busybox-1.18.5.orig/init/init.c busybox-1.18.5/init/init.c
--- busybox-1.18.5.orig/init/init.c	2011-06-12 09:41:51.000000000 -0600
+++ busybox-1.18.5/init/init.c	2011-07-27 14:23:18.381240460 -0600
@@ -113,7 +113,8 @@
 #include <paths.h>
 #include <sys/resource.h>
 #ifdef __linux__
-#include <linux/vt.h>
+# include <linux/vt.h>
+# include <sys/sysinfo.h>
 #endif
 #if ENABLE_FEATURE_UTMP
 # include <utmp.h> /* DEAD_PROCESS */
diff -Naur busybox-1.18.5.orig/procps/free.c busybox-1.18.5/procps/free.c
--- busybox-1.18.5.orig/procps/free.c	2011-06-12 09:41:52.000000000 -0600
+++ busybox-1.18.5/procps/free.c	2011-07-27 14:23:18.381240460 -0600
@@ -10,6 +10,9 @@
 /* getopt not needed */
 
 #include "libbb.h"
+#ifdef __linux__
+# include <sys/sysinfo.h>
+#endif
 
 struct globals {
 	unsigned mem_unit;
diff -Naur busybox-1.18.5.orig/procps/ps.c busybox-1.18.5/procps/ps.c
--- busybox-1.18.5.orig/procps/ps.c	2011-06-12 09:41:52.000000000 -0600
+++ busybox-1.18.5/procps/ps.c	2011-07-27 14:23:18.381240460 -0600
@@ -16,6 +16,9 @@
 
 #if ENABLE_DESKTOP
 
+#ifdef __linux__
+# include <sys/sysinfo.h>
+#endif
 #include <sys/times.h> /* for times() */
 #ifndef AT_CLKTCK
 #define AT_CLKTCK 17
diff -Naur busybox-1.18.5.orig/procps/uptime.c busybox-1.18.5/procps/uptime.c
--- busybox-1.18.5.orig/procps/uptime.c	2011-06-12 09:41:52.000000000 -0600
+++ busybox-1.18.5/procps/uptime.c	2011-07-27 14:23:18.385240384 -0600
@@ -16,6 +16,10 @@
 /* getopt not needed */
 
 #include "libbb.h"
+#ifdef __linux__
+# include <sys/sysinfo.h>
+#endif
+
 
 #ifndef FSHIFT
 # define FSHIFT 16              /* nr of bits of precision */
