diff -Naur linux-2.6.38.2.orig/arch/x86/boot/Makefile linux-2.6.38.2/arch/x86/boot/Makefile
--- linux-2.6.38.2.orig/arch/x86/boot/Makefile	2011-03-27 12:37:20.000000000 -0600
+++ linux-2.6.38.2/arch/x86/boot/Makefile	2011-07-26 22:10:08.312027049 -0600
@@ -88,7 +88,7 @@
 
 SETUP_OBJS = $(addprefix $(obj)/,$(setup-y))
 
-sed-voffset := -e 's/^\([0-9a-fA-F]*\) . \(_text\|_end\)$$/\#define VO_\2 0x\1/p'
+sed-voffset := -e 's/^\([0-9a-fA-F]*\) . \(_text\)$$/\#define VO_\2 0x\1/p' -e 's/^\([0-9a-fA-F]*\) . \(_end\)$$/\#define VO_\2 0x\1/p'
 
 quiet_cmd_voffset = VOFFSET $@
       cmd_voffset = $(NM) $< | sed -n $(sed-voffset) > $@
@@ -97,7 +97,10 @@
 $(obj)/voffset.h: vmlinux FORCE
 	$(call if_changed,voffset)
 
-sed-zoffset := -e 's/^\([0-9a-fA-F]*\) . \(startup_32\|input_data\|_end\|z_.*\)$$/\#define ZO_\2 0x\1/p'
+sed-zoffset := -e 's/^\([0-9a-fA-F]*\) . \(startup_32\)$$/\#define ZO_\2 0x\1/p' \
+	-e 's/^\([0-9a-fA-F]*\) . \(input_data\)$$/\#define ZO_\2 0x\1/p' \
+	-e 's/^\([0-9a-fA-F]*\) . \(_end\)$$/\#define ZO_\2 0x\1/p' \
+	-e 's/^\([0-9a-fA-F]*\) . \(z_.*\)$$/\#define ZO_\2 0x\1/p'
 
 quiet_cmd_zoffset = ZOFFSET $@
       cmd_zoffset = $(NM) $< | sed -n $(sed-zoffset) > $@
